### DISCUSSION TITLE
FCS imputation specify different ignored rows for different imputation models

### DISCUSSION THREAD

**Discussion by tinneuro** [amices/mice#595] [Getting started]
Hi,

I want to impute missing data for two variables of my dataset with 10 colums/variables. For these two variables I use different imputation models by specifying the formula used for each of the variables. Thus for the two variables x9 andx10 I have individual differing formulas for the imputation. i.e. formula for x9~ x1 + x2 + x3 and x10~ x 1+ x4 + x5  where we can see that x9 is not a predictor of x10 and x10 is not used as a predictor for x9. 
In addition to the different formulas I also want to train the two imputation models on different rows of the dataset where I would use the ignore functionality of mice. But with the ignore I can only specify one vector of TRUE and FALSE which would then be the same for both models (i.e. x9 and x10) but I want the models to be trained on differing rows of the data. How can I achieve that? Maybe I could solve that with blocks but I did not find more guidance about how to solve this. Thanks a lot!

**Comment by stefvanbuuren** [comment]
Did you consider splitting the data into two sets: `{x1, x2, x3, x9}` and `{x1, x4, x5, x10}`, and impute these separately with their own `ignore` vector?

**Reply by tinneuro** [reply]
Dear Stefan,
thanks a lot for your quick reply. Yes this is exactly what I did initially. But I need a combined dataset to compute joint statistics and also for regression models where I want to have all imputed data together because I want to put x9 and x10 together with other variables in the model.

**Reply by tinneuro** [reply]
The other option I saw when having two seperate imputed datasets was then to join the obtained imputed values for x10 to the first imputed dataset. But for this I probably need to transform with complete and then again as.mids(). Is this an option which could work?

**Reply by stefvanbuuren** [reply]
`mice::cbind.mids()` combines the two `mids` objects into one. Give that a try.

**Reply by tinneuro** [reply]
Thanks! The problem is that I actually need to join the two imputed datasets by an Id. Thus I am now considering to rather use mice::rbind.mids() and have a long data format and then postprocess with complete and pivot_wider() to a wide format and transform back to a mids object. Unfortunately this is tedious...

**Reply by stefvanbuuren** [reply]
Yes. I see. `cbind.mids()` requires the same number of rows in both datasets. Otherwise, you're in the woods, indeed. Good luck.

**Reply by tinneuro** [reply]
Haha. Yes I also realised that. But still thanks a lot for your feedback it helped me!
